iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/reports/tflint/TFLintImporter.java

```diff
@@ -0,0 +1,165 @@
+    if (rule == null) {
+      // it's error, not an issue
+      String severity = (String) issueJson.get("severity");
+
+      NewExternalIssue externalIssue = context.newExternalIssue()
+        .ruleId("tflint.error")
+        .type(RuleType.CODE_SMELL)
+        .engineId("tflint")
+        .severity(severity(severity))
+        .remediationEffortMinutes(0L);
+      externalIssue.at(errorLocation(issueJson, externalIssue));
+      return externalIssue;
+    }
+
+    String ruleId = (String) rule.get("name");
+    String severity = (String) rule.get("severity");
+
+    NewExternalIssue externalIssue = context.newExternalIssue()
+      .ruleId(ruleId)
+      .type(RuleType.CODE_SMELL)
+      .engineId("tflint")
+      .severity(severity(severity))
+      .remediationEffortMinutes(0L);
+    externalIssue.at(issueLocation(issueJson, externalIssue));
+    return externalIssue;
+  }
+
+  private NewIssueLocation issueLocation(JSONObject issueJson, NewExternalIssue externalIssue) {
+    JSONObject range = (JSONObject) issueJson.get("range");
+    String filename = (String) range.get("filename");
+
+    InputFile inputFile = inputFile(filename);
+
+    JSONObject start = (JSONObject) range.get("start");
+    int startLine = asInt(start.get("line"));
+    int startColumn = asInt(start.get("column"));
+    JSONObject end = (JSONObject) range.get("end");
+    int endLine = asInt(end.get("line"));
+    int endColumn = asInt(end.get("column"));
+    TextRange textRange = inputFile.newRange(startLine, startColumn - 1, endLine, endColumn - 1);
+
+    String message = (String) issueJson.get("message");
+
+    return externalIssue.newLocation()
+      .message(message)
+      .on(inputFile)
+      .at(textRange);
+  }
+
+  private NewIssueLocation errorLocation(JSONObject issueJson, NewExternalIssue externalIssue) {
+    String message = (String) issueJson.get("message");
+    Matcher matcher = FILENAME_PATTERN.matcher(message);
+    if (!matcher.find()) {
+      throw new ReportImporterException("Can't extract filename from error message");
+    }
+    String filename = matcher.group(1);
+    int startLine = Integer.parseInt(matcher.group(2));
+    int startColumn = Integer.parseInt(matcher.group(3));
+    int endLineOffset = Integer.parseInt(matcher.group(4));
+
+    InputFile inputFile = inputFile(filename);
+
+    TextRange textRange;
+    try {
+      textRange = inputFile.newRange(startLine, startColumn - 1, startLine, endLineOffset - 1);
+    } catch (IllegalArgumentException e) {
+      textRange = inputFile.selectLine(startLine);
+    }
+
+    return externalIssue.newLocation()
+      .message(message)
+      .on(inputFile)
+      .at(textRange);
+  }
+
+  private static Severity severity(String severity) {
+    String text = severity.toUpperCase(Locale.ENGLISH);
+    if ("WARNING".equals(text)) {
+      text = "MINOR";
+    }
+    if ("ERROR".equals(text)) {
+      text = "BLOCKER";
+    }
+    return Severity.valueOf(text);
```

Comment:
See comment in `AbstractJsonReportImporter`. Could produce `IllegalArgumentException` which isn't catched.